File: problem-test/src/main/java/com/dudgns/problemtest/entity/id/CarCategoryMappingId.java
package com.dudgns.problemtest.entity.id;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.*;

import java.io.Serializable;
import java.util.UUID;

@Embeddable
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class CarCategoryMappingId implements Serializable {
    @Column(name = "car_idx", nullable = false, updatable = false)
    private Long carIdx;
    @Column(name = "category_type", nullable = false, updatable = false)
    private Integer categoryType;
}

File: problem-test/commons/project-dudgns-data-entity/src/main/kotlin/dudgns/com/backend/commons/data/entity/problemTest/id/CarCategoryMappingId.kt
package dudgns.com.backend.commons.data.entity.problemTest.id

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import java.io.Serializable

@Embeddable
class CarCategoryMappingId : Serializable {
    @Column(name = "car_idx", nullable = false, updatable = false)
    var carIdx: Long? = null

    @Column(name = "category_type", nullable = false, updatable = false)
    var categoryType: Int? = null
}